lib/addons/ebs-csi-driver/index.ts

```diff
@@ -0,0 +1,62 @@
+import { CfnAddon } from "@aws-cdk/aws-eks";
+import {ClusterAddOn} from "../..";
+import { ClusterInfo } from "../../spi";
+import {AmazonEksEbsCsiDriverPolicy} from "./iam-policy";
+import * as iam from "@aws-cdk/aws-iam";
+
+/**
+ * User provided overrides for the add-on
+ */
+export class EbsCsiDriveAddOnProps {
+    /**
+     * Version of the Helm chart to be installed
+     */
+    readonly version?: string;
+}
+
+/**
+ * Default values for the add-on
+ */
+const defaultProps = {
+    addOnName: 'aws-ebs-csi-driver',
+    namespace: 'kube-system',
+    version: 'v1.4.0-eksbuild.preview'
+};
+
+/**
+ * Implementation of EBS CSI Driver EKS add-on.
+ */
+export class EbsCsiDriverAddOn implements ClusterAddOn {
+
+    readonly options: EbsCsiDriveAddOnProps;
+
+    constructor(ebsCsiDriveAddOnProps?: EbsCsiDriveAddOnProps) {
+        this.options = {...defaultProps, ...ebsCsiDriveAddOnProps };
+    }
+
+    deploy(clusterInfo: ClusterInfo): void {
+        const cluster = clusterInfo.cluster;
+
+        // Permissions
+
+        const serviceAccount = cluster.addServiceAccount(defaultProps.addOnName, {
+            name: defaultProps.addOnName,
+            namespace: defaultProps.namespace,
+        });
+
+        AmazonEksEbsCsiDriverPolicy.Statement.forEach((statement) => {
+            serviceAccount.addToPrincipalPolicy(iam.PolicyStatement.fromJson(statement));
+        });
```

Comment:
Lets move the IRSA creation to `core-addon.ts` and add another property to the interface to accept `PolicyStatement[]`. This way its available for the next EKS managed addon that needs an IRSA role and also lets us use `extends` instead of implementing the `ClusterAddOn` interface here.

```ts
constructor(version?: string) {
    super({
        addOnName: "aws-ebs-csi-driver",
        version: version ?? "v1.4.0-eksbuild.preview"
     });
}
```
@shapirov103 thoughts?

Also should we take this opportunity to provide for a `ConfigMap` property on the interface that lets users alter the fields in the add-on configuration?

Reply:
It makes sense to centralize IRSA for core addons, although the approach for vpc-cni was a bit different and we did not manage to use SA when add-on was added after the cluster has been already provisioned. I assume EBS add-on may not fall into the same bucket. 
If anything, you can leverage our `sa-utils.ts`, we have `createServiceAccount` there as well.